### PR TITLE
ocrd-network: optional deploy mongodb and queue

### PR DESCRIPTION
**Pull request by joschrew**
Just a draft for now for easier usage.

This is an extensions to [1030](https://github.com/OCR-D/core/pull/1030).

With this mongoDB and rabbitMQ can be skipped with the deployer to allow self management of this two parts/containers.
For the deployment on the VM I want to change to self managed queue and database so I need this therefore.

### CODE REVIEW

**Comment by bertsky** [comment]
> This is an extensions to [1030](https://github.com/OCR-D/core/pull/1030).

please rebase onto #1030:

    git fetch origin
    git rebase origin/processor-server
    git push -f

**Comment by MehmedGIT** [comment]
He means to rebase to #1030, not the master branch. So it's easier to see what has been changed on top of #1030.

**Comment by joschrew** [comment]
How to do that?

File: ocrd_network/ocrd_network/deployer.py [hunk 1]
```diff
@@ -229,6 +231,23 @@ def deploy_rabbitmq(
...
+                self.data_queue.username,
+                self.data_queue.password
+            )
+            wait_for_rabbitmq_availability(
+                host=self.data_queue.address,
+                port=int(self.data_queue.port),
+                vhost='/',
+                username=self.data_queue.username,
+                password=self.data_queue.password
+            )
```
Comment (MehmedGIT) [inline ocrd_network/ocrd_network/deployer.py:249]: Why do we need both of these availability checks? Isn't `wait_for_rabbitmq_availability` already performing the required check? I think it does. I like the new naming convention - `verify_rabbitmq_available`. It just has to do multiple checks to simulate the timeout just like in `wait_for_rabbitmq_availability`. Then we can get rid of the `wait_for_rabbitmq_availability` method.

Comment (joschrew) [reply ocrd_network/ocrd_network/deployer.py]: Thanks, updated accordingly

File: ocrd_network/ocrd_network/deployment_utils.py [hunk 2]
```diff
@@ -101,7 +103,16 @@ def wait_for_rabbitmq_availability(
         else:
             # TODO: Disconnect the dummy_publisher here before returning...
             return
-    raise RuntimeError('Error waiting for queue startup: timeout exceeded')
+    raise RuntimeError(f'Cannot connet to Rabbitmq host: {host}, port: {port}, '
```
Comment (MehmedGIT) [inline ocrd_network/ocrd_network/deployment_utils.py:106]: Typo: connet -> connect. Also, `RabbitMQ`.

File: ocrd_network/ocrd_network/deployment_utils.py [hunk 3]
```diff
@@ -101,7 +103,16 @@ def wait_for_rabbitmq_availability(
...
+                       f'vhost: {vhost}, username: {username}')
+
+
+def verify_mongodb_available(mongo_url: str) -> None:
+    try:
+        client = MongoClient(mongo_url, serverSelectionTimeoutMS=1000.0)
+        client.admin.command("ismaster")
+    except Exception:
+        raise RuntimeError(f'Cannot connet to MongoDb: {re.sub(r":[^@]+@", ":****@", mongo_url)}')
```
Comment (MehmedGIT) [inline ocrd_network/ocrd_network/deployment_utils.py:115]: Typo: connet -> connect. Also, `MongoDB` (in other places as well).

**Review by MehmedGIT — COMMENTED** [review]
Looks good to me and can be merged.

File: ocrd_network/ocrd_network/deployment_utils.py (hunk 2 above)
Comment (joschrew) [reply ocrd_network/ocrd_network/deployment_utils.py]: Thanks, fixed

File: ocrd_network/ocrd_network/deployment_utils.py (hunk 3 above)
Comment (joschrew) [reply ocrd_network/ocrd_network/deployment_utils.py]: Thanks, fixed as well

**Pull request merged**